apps/hash-graph/lib/graph/src/api/rest/entity_type.rs

```diff
@@ -105,12 +129,70 @@ async fn create_entity_type<P: StorePool + Send>(
     body: Json<CreateEntityTypeRequest>,
     // TODO: We want to be able to return `Status` here we should try and create a general way to
     //  call `status_to_response` for our routes that return Status
-) -> Result<Json<ListOrValue<OntologyElementMetadata>>, Response> {
-    let Json(CreateEntityTypeRequest {
+) -> Result<Json<ListOrValue<OntologyElementMetadata>>, Response>
+where
+    for<'pool> P::Store<'pool>: RestApiStore,
+{
+    let mut store = pool.acquire().await.map_err(|report| {
+        tracing::error!(error=?report, "Could not acquire store");
+        status_to_response(Status::new(
+            hash_status::StatusCode::Internal,
+            Some(
+                "Could not acquire store. This is an internal error, please report to the \
```

Comment:
Any reason for this pool acquisition using `hash_status` when the `data_type` and `property_type` equivalent changes don't?

Reply:
When the status library was introduced it seems that these methods were not adjusted, most probably due to the reason that the status lib in its current form is extremely boiler-platy and hard to use. As flagged offline I'm going to revisit error handling in the Rest API soon, so I chose to leave the behavior as it is right now.